src/terrain/terrain_helper.cpp

```diff
@@ -0,0 +1,23 @@
+#include "terrain_helper.hpp"
+#include "terrain.hpp"
+#include "tile.hpp"
+
+namespace terrain {
+namespace helper {
+
+Tile* Ninos_horrible_function(Terrain& ter, Tile*tile, int x, int y){
```

Comment:
How about `get_adjacent_tile`

And look at how much nicer the code is now that you've removed duplication.

1. You don't have to grok it twice
2. If you assume its correct now you don't even have to bother reading it

Also you need to re-run your formatter.